backend/module/chaos/serverexperimentation/xds/ecds.go

```diff
@@ -0,0 +1,246 @@
+package xds
+
+import (
+	"fmt"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/golang/protobuf/ptypes/duration"
+	"sync"
+	"time"
+
+	gcpCoreV3 "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	gcpRoute "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	gcpFilterCommon "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/common/fault/v3"
+	gcpFilterFault "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/fault/v3"
+	gcpType "github.com/envoyproxy/go-control-plane/envoy/type/v3"
+	gcpTypes "github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"go.uber.org/zap"
+
+	proto "github.com/golang/protobuf/proto"
+
+	experimentation "github.com/lyft/clutch/backend/api/chaos/experimentation/v1"
+	serverexperimentation "github.com/lyft/clutch/backend/api/chaos/serverexperimentation/v1"
+)
+
+const (
+	HeaderXEnvoyDownstreamServiceCluster = `x-envoy-downstream-service-cluster`
+	ExternalFaultFilterConfigName        = `envoy.egress.extension_config.%s`
+	InternalFaultFilterConfigName        = `envoy.extension_config`
+	FaultFilterTypeURL                   = `type.googleapis.com/envoy.extensions.filters.http.fault.v3.HTTPFault`
+
+	DelayPercentRuntime    = "%s.http.delay.percentage"
+	DelayDurationRuntime   = "%s.http.delay.fixed_duration_ms"
+	AbortHttpStatusRuntime = "%s.http.abort.http_status"
+	AbortPercentRuntime    = "%s.http.abort.abort_percent"
+)
+
+// Default abort and delay fault configs
+var DefaultAbortFaultConfig = &gcpFilterFault.FaultAbort{
+	ErrorType: &gcpFilterFault.FaultAbort_HttpStatus{
+		HttpStatus: 503,
+	},
+	Percentage: &gcpType.FractionalPercent{
+		Numerator:   0,
+		Denominator: gcpType.FractionalPercent_HUNDRED,
+	},
+}
+
+var DefaultDelayFaultConfig = &gcpFilterCommon.FaultDelay{
+	FaultDelaySecifier: &gcpFilterCommon.FaultDelay_FixedDelay{
+		FixedDelay: &duration.Duration{
+			Seconds: 0,
+		},
+	},
+	Percentage: &gcpType.FractionalPercent{
+		Numerator:   0,
+		Denominator: gcpType.FractionalPercent_HUNDRED,
+	},
+}
+
+type ECDSConfig struct {
+	enabledClusters []string
+
+	faultRuntimePrefix string
+
+	requestedResourcesMap sync.Map
```

Comment:
Looking at this I think this might not work because this would be a copy of the map used by the callbacks? An integration test that this works right would be awesome